thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java

```diff
@@ -205,12 +207,15 @@ public void purge(@Nullable final Duration expiryDurationOptional,
 
     final long startTime = System.nanoTime();
     final List<TaskDTO> tasksToBeDeleted = filter(new DaoFilter()
+        // non locking read
+        .setTransactionIsolationLevel(Connection.TRANSACTION_READ_UNCOMMITTED)
```

Comment:
tasks purge is a core operation and not some metric collection. Right?

Are we doing this to speed up this process 
because even if some tasks are uncommitted, they'll be purged in next cycle properly anyways?

Reply:
this query is doing a full scan (createTime index will be used but logically it's still a full scan), so it would prevent inserts and updates on the table 
so it blocks the 
this means the 

this query is still pretty bad because it loads 5000 tasks in memory, just to delete them. 
but I'll fix this later.